src/adapter/http/src/middleware/current_account_resolver_layer.rs

```diff
@@ -0,0 +1,136 @@
+// Copyright Kamu Data, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use std::pin::Pin;
+use std::task::{Context, Poll};
+
+use axum::body::Body;
+use axum::http::{Request, StatusCode};
+use axum::response::Response;
+use axum::RequestExt;
+use futures::Future;
+use kamu::domain::{auth, CurrentAccountSubject};
+use tower::{Layer, Service};
+
+use crate::AccessToken;
+
+/////////////////////////////////////////////////////////////////////////////////
+
+#[derive(Debug, Clone)]
+pub struct CurrentAccountResolverLayer {}
+
+impl CurrentAccountResolverLayer {
+    pub fn new() -> Self {
+        Self {}
+    }
+}
+
+impl<Svc> Layer<Svc> for CurrentAccountResolverLayer {
+    type Service = CurrentAccountResolverMiddleware<Svc>;
+
+    fn layer(&self, inner: Svc) -> Self::Service {
+        CurrentAccountResolverMiddleware { inner }
+    }
+}
+
+/////////////////////////////////////////////////////////////////////////////////
+
+#[derive(Debug, Clone)]
+pub struct CurrentAccountResolverMiddleware<Svc> {
+    inner: Svc,
+}
+
+impl<Svc> CurrentAccountResolverMiddleware<Svc> {
```

Comment:
Would `AuthenticationMiddleware` be a better name for this?

Reply:
Renamed middleware and layer